segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/MetadataStore.java

```diff
@@ -149,7 +155,13 @@ public void close() {
         }
 
         ArrayView segmentInfo = SegmentInfo.serialize(SegmentInfo.newSegment(segmentName, segmentType, attributes));
-        CompletableFuture<Void> result = createSegment(segmentName, segmentInfo, new TimeoutTimer(timeout));
+        CompletableFuture<Void> result;
+        if (segmentType != SegmentType.TRANSIENT_SEGMENT) {
+            result = createSegment(segmentName, segmentInfo, new TimeoutTimer(timeout));
+        } else {
+            result = CompletableFuture.runAsync(() -> submitAssignment(newSegment(segmentName, segmentType, attributes), false, timeout));
```

Comment:
1. Please factor this out into its own method.
2. Why `.runAsync`? Isn't `submitAssignment` returning a future? The way you wrote it, you'll return a CompletableFuture that will almost immediately complete - since all it does is initiate `submitAssignment` but not await its completion. Please check the Javadoc for `CompletableFuture` to understand how the methods you're using work.

Reply:
1. Okay
2. It is, but is returning a `Long` and not `Void`. I didn't believe you could directly cast the types.